packages/graphql/tests/integration/issues/6005-filters.int.test.ts

```diff
@@ -0,0 +1,281 @@
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { UniqueType } from "../../utils/graphql-types";
+import { TestHelper } from "../../utils/tests-helper";
+
+/**
+ * For the following tests we assuming that the deprecated syntax keep the existing behavior while when using the new syntax the
+ * distinct is applied. This is applied only to count aggregation as for node aggregations second thoughts are needed.
+ **/
+describe("https://github.com/neo4j/graphql/issues/6005 filters", () => {
+    let Movie: UniqueType;
+    let Actor: UniqueType;
+
+    const testHelper = new TestHelper();
+
+    beforeEach(async () => {
+        Movie = testHelper.createUniqueType("Movie");
+        Actor = testHelper.createUniqueType("Actor");
+
+        const typeDefs = /* GraphQL */ `
+            type ${Movie} @node {
+                title: String!
+                actors: [${Actor}!]! @relationship(type: "ACTED_IN", direction: IN, properties: "ActedIn")
+            }
+            type ${Actor} @node {
+                name: String!
+                age: Int!
+                born: DateTime!
+                movies: [${Movie}!]! @relationship(type: "ACTED_IN", direction: OUT, properties: "ActedIn")
+            }
+            type ActedIn @relationshipProperties {
+                screentime: Int!
+                character: String!
+            }
+        `;
+
+        await testHelper.initNeo4jGraphQL({ typeDefs });
+        // Arnold as two edge with terminator and two actors have the same age to test that the distinct is not applied on properties
+        await testHelper.executeCypher(`
+            CREATE (m:${Movie} { title: "Terminator"})
+            CREATE (m)<-[:ACTED_IN { screentime: 60, character: "Terminator" }]-(arnold:${Actor} { name: "Arnold", age: 54, born: datetime('1980-07-02')})
+            CREATE (m)<-[:ACTED_IN { screentime: 120, character: "Sarah" }]-(:${Actor} {name: "Linda", age: 37, born: datetime('2000-02-02')})
+            CREATE (m)<-[:ACTED_IN { screentime: 120, character: "Another Character" }]-(:${Actor} {name: "Another actor", age: 37, born: datetime('2000-02-02')})
+            CREATE (m)<-[:ACTED_IN { screentime: 10, character: "Future Terminator" }]-(arnold)
+        `);
+    });
+
+    afterEach(async () => {
+        await testHelper.close();
+    });
+
+    test("should filter movies by actors count with duplicate results (deprecated syntax)", async () => {
```

Comment:
We should move the deprecated syntax to the `deprecated` folder in a separate test file `6005.int.test`

Reply:
In this case, I put it in the same file intentionally to highlight the differences between the two, not only in the API but also in the behavior. I got the feeling that by moving in two separate files, a developer without the correct context could identify one of the behaviors as a regression or a bug.

Reply:
I think that the same issue number being linked with the file in `deprecated` is good context on that.

In any case, deprecated behaviours are that, deprecated, regressions on that are tested in the deprecated folder. The new one is already in the correct place.

I'm just thinking that now that we finally start to have some conventions in our tests, it is not worth breaking those just for a quite edge case confusion that may arise with some deprecated behaviour. Breaking convention will generate confusion